app/controllers/api/v1/search_controller.rb

```diff
@@ -757,9 +789,8 @@ def self.get_facet_match_for_tdr_result(facet, result_row)
         else
           matches = []
           facet[:filters].each do |filter|
-            # look for match on the column name, and see which filter value also matched (ontology label or id)
-            matches << result_row.each_pair.select { |col, val| col == tdr_name && (val == filter[:name] || filter[:id]) }
-                                 .flatten
+            matches << result_row.each_pair.select { |col, val| col == tdr_name && (val == filter[:name] || val == filter[:id] ) }
```

Comment:
Blocker: this fixes the problem we saw yesterday with bogus matches, however it now throws an error if you search for other facets that don't match on the Pulmonary TDR study:
```
undefined method `to_sym' for nil:NilClass
single_cell_portal_core/lib/facet_name_converter.rb:70:in `convert_to_model'
single_cell_portal_core/app/controllers/api/v1/search_controller.rb:644:in `block (3 levels) in simplify_tdr_facet_search_results'
single_cell_portal_core/app/controllers/api/v1/search_controller.rb:642:in `each_pair'
single_cell_portal_core/app/controllers/api/v1/search_controller.rb:642:in `block (2 levels) in simplify_tdr_facet_search_results'
```
The issue is that we now end up with a "match" that looks like this: `{nil => nil}`, which throws the `nil` error when trying to convert from TIM to Alexandria.  If we add the following line right before the final return of `matches`, it will work, though we still have the problem identified in SCP-3544 where our matching is too lax (out of scope for this):
```
matches.reject! { |key, value| key.blank? || value.blank? }
```

Reply:
oh shoot good catch, thanks!